src/components/organisms/SearchBar/SearchBar.js

```diff
@@ -0,0 +1,134 @@
+import { Input } from "components/atoms/Input/Input";
+import { CryptoApiContext } from "providers/CryptoApiProvider";
+import React, { useContext, useEffect, useState } from "react";
+import { Wrapper } from "./SearchBar.styles";
+import styled from "styled-components";
+import useModal from "components/organisms/Modal/useModal";
+import debounce from "lodash.debounce";
+import CoinDetails from "components/molecules/CoinDetails/CoinDetails";
+
+const InfoWrapper = styled.div`
+  flex-basis: 20%;
+  padding: 20px;
+  span {
+    font-weight: bold;
+    font-size: ${({ theme }) => theme.fontSize.xl};
+  }
+`;
+
+const SearchWrapper = styled.div`
+  flex-grow: 1;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  position: relative;
+  padding: 50px;
+
+  input {
+    width: 100%;
+  }
+`;
+
+const SearchResults = styled.ul`
+  position: absolute;
+  width: 90%;
+  max-height: 400px;
+  overflow-y: scroll;
+  background-color: blue;
+  top: 55%;
+  z-index: 111;
+  border-radius: 20px;
+  background-color: ${({ theme }) => theme.colors.lightPurple};
+  list-style: none;
+  padding-top: 20px;
+  padding: 10px;
+
+  li {
+    display: flex;
+    justify-content: flex-start;
+    align-items: center;
+    padding-left: 15px;
+    height: 40px;
+    border-radius: 10px;
+    margin: 10px;
+    background-color: #fff;
+    cursor: pointer;
+
+    img {
+      height: 70%;
+      margin: 0 20px 0 0;
+    }
+
+    span {
+      font-weight: bold;
+      margin-left: 5px;
+    }
+  }
+`;
+
+const SearchBar = () => {
+  const [matchingCoins, setMatchingCoins] = useState([]);
+  const [searchingCoin, setSearchingCoin] = useState("");
+  const { Modal, isOpen, handleOpenModal, handleCloseModal } = useModal();
+  const { handleSearchCoin, getCoinValue } = useContext(CryptoApiContext);
+  const [currentCoin, setCurrentCoin] = useState(null);
```

Comment:
tutaj też lubie trzymać porządek, u mnie by to wyglądało tak:

```
const { Modal, isOpen, handleOpenModal, handleCloseModal } = useModal();
const { handleSearchCoin, getCoinValue } = useContext(CryptoApiContext);

const [matchingCoins, setMatchingCoins] = useState([]);
const [searchingCoin, setSearchingCoin] = useState("");
const [currentCoin, setCurrentCoin] = useState(null);
```

i cały czas zachęcam do typescripta, cały ten kod który tutaj masz napisany będzie działał w typescripcie, a jednak typowanie to jest ważna sprawa 

w TS będzie to wyglądało w ten sposób:
```
type Coin = "ETH" | "BTC" | "SOL" | "LUNA";

const { Modal, isOpen, handleOpenModal, handleCloseModal } = useModal();
const { handleSearchCoin, getCoinValue } = useContext(CryptoApiContext);

const [matchingCoins, setMatchingCoins] = useState<Coin[]>([]);
const [searchingCoin, setSearchingCoin] = useState<Coins | null>(null);
const [currentCoin, setCurrentCoin] = useState<Coin | null>(null);
```